src/main/java/AddCommand.java

```diff
@@ -0,0 +1,81 @@
+import java.io.IOException;
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+
+/**
+ * The AddCommand class is used to update the current TaskList with the newly added task
+ * and prints the added task to the user.
+ */
+class AddCommand extends Command {
+    protected String command;
+
+    AddCommand(String command) {
+        this.command = command;
+    }
+
+    /**
+     * The execute method of the AddCommand is used to add the task in the command given into
+     * the TaskList and updates the stored TaskList in duke.txt.
+     *
+     * @param tasks This is the current TaskList in duke.txt.
+     * @param ui The current Ui created.
+     * @param storage This is used for saving of the new TaskList into duke.txt.
+     * @throws IOException if file input cannot be written to or closed.
+     */
+    @Override
+    void execute(TaskList tasks, Ui ui, Storage storage) throws IOException {
+        try {
+            if (this.command.startsWith("deadline")) {
+                String commandWithoutDeadline = this.command.substring(9);
+                String[] commands = commandWithoutDeadline.split("/by");
+                LocalDate localDate = LocalDate.parse(commands[1].trim());
+                Deadline deadline = new Deadline(commands[0], localDate);
+                tasks.getTaskList().add(deadline);
+                storage.save(tasks);
+
+                System.out.println("\n" + "Alright, I've added this task:" + "\n");
```

Comment:
Hey for your system outputs, maybe using String.format would be a cleaner way of injecting variables into the string. Also you can use %n for String.format so that the newline character generated follows the OS java is running on